onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -367,9 +367,9 @@ def __init__(self, path_or_bytes, sess_options=None, providers=None, provider_op
 
         try:
             self._create_inference_session(providers, provider_options, disabled_optimizers)
-        except ValueError:
+        except ValueError as e:
             if self._enable_fallback:
-                print(f"EP Error using {providers}")
+                print(f"EP Error {e} when using {providers}")
```

Comment:
Should we emit a warning instead?

Reply:
Good idea, but let us discuss a bit more before changing the default behavior and do it in another PR if needed.
This change is more to try and merge https://github.com/microsoft/onnxruntime/pull/15012